business_objects/record.py

```diff
@@ -88,6 +91,48 @@ def get_ids_of_manual_records_by_labeling_task(
     )
 
 
+def count_records_without_tokenization(project_id: str) -> int:
+    query = f"""
+    SELECT count(r.id) c
+    FROM ({get_records_without_tokenization(project_id, 0, True)}) r
+    """
+    return general.execute(query).first().c
+
+
+def get_attribute_data_with_doc_bins_of_records(
+    project_id: str, attribute_name: str
+) -> List[Any]:
+    query = f"""
+    SELECT rt.id, r."data" ->> '{attribute_name}' "attribute_data", rt.bytes
+    from record r
+    JOIN record_tokenized rt
+    ON  r.id = rt.record_id
+    WHERE r.project_id = '{project_id}'
+    AND rt.project_id = '{project_id}'
+    """
+    return general.execute_all(query)
+
+
+def update_bytes_of_record_tokenized(values: List[Dict[str, Any]]) -> None:
+    query = (
+        update(RecordTokenized)
+        .where(RecordTokenized.id == bindparam("_id"))
```

Comment:
i think we should still include the project id for indexing purposes

Reply:
In theory i would do it but for this query i would leave it as it is since only checking for the project_id as second where condition just created errors for me

Reply:
not sure what you mean, i tested with:
```
project_id = "9221e93a-e8d1-4fcb-b4a7-967cc7d13b21"
    query = (
        update(RecordTokenized)
        .where(
            RecordTokenized.id == bindparam("_id"),
            RecordTokenized.project_id == project_id,
        )
        .values({"bytes": bindparam("bytes")})
    )
```

and didn't receive any errors (project id of course would be a parameter instead) -> maybe I'm missing something though

Reply:
strange, maybe i am missing something, but for me i think i got problems here, but anyways it was the not working version from before, so yes lets take the project id in here for indexing if it is just the code shown above (project id as parameter ofc)

Reply:
resolved